### PR TITLE
astrometry-net 0.76 (new formula)

### PR DESCRIPTION
**Pull request by dstndstn**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by dstndstn** [comment]
This package was among packages considered "notable enough" to warrant migration in #6331.  Thx!

File: Formula/astrometry-net.rb
Comment (dstndstn) [reply Formula/astrometry-net.rb]: sorry, *what* doesn't exist in core?  Yeah, this package doesn't exist in core; like I said, it used to be in the homebrew-science tap and I would like to migrate it to homebrew-core.  All the dependencies do seem to exist in homebrew-core, if that's what you were indicating.

Comment (commitay) [reply Formula/astrometry-net.rb]: `wcslib` isn't a formula.

Comment (dstndstn) [reply Formula/astrometry-net.rb]: Ohhh, I'm sorry, I'm wrong, indeed "wcslib" does not exist yet!  Will send a PR for it.

**Comment by dstndstn** [comment]
Depends on #30065 
Sorry for the trouble; not sure why *git status* didn't tell me that wcslib.rb only existed in my local directory!

**Comment by MikeMcQuaid** [comment]
@BrewTestBot test this please

**Comment by zbeekman** [comment]
@BrewTestBot test this please

(so I can see the logs which have gone missing)

**Comment by dstndstn** [comment]
@BrewTestBot test this please

**Comment by dstndstn** [comment]
@BrewTestBot test this please (again)

**Comment by dstndstn** [comment]
@BrewTestBot test this please, because your last try borked

**Comment by dstndstn** [comment]
@fxcoudert , @zbeekman , @DomT4 : would you be willing to help with this one too?

audit is clean, build seems to work.

**Comment by zbeekman** [comment]
@dstndstn Yes, it is on my todo list. I'll take a quick look now

**Review by zbeekman — CHANGES_REQUESTED** [review]
My only concern here is the extensive dependency list, and in particular whether `wget` is truly a runtime dependency and the rationale for using it, if it is.

If it really, truly is a runtime dependency, I would appreciate it if you could help me understand why. If it is actually a build dependency, then we need to understand what resources it is fetching at build time and add a `=> :build` modifier to the dependency.

Other than this, this formula looks good to me. (I will reserve comment on Makefile only build systems, however 😉)

File: Formula/astrometry-net.rb
Comment (dstndstn) [reply Formula/astrometry-net.rb]: wget is an honest runtime dependency, called from the C code if the user gives an http:// or ftp:// URL as an input argument.

**Comment by dstndstn** [comment]
@zbeekman  thanks for your attention :)

As for Makefiles... well, this package builds on OSX, FreeBSD, Linux, and Solaris, so there's that...

File: Formula/astrometry-net.rb
Comment (dstndstn) [reply Formula/astrometry-net.rb]: (but don't ask me to defend *why* it should accept URLs as inputs... I'm sure It Seemed Like A Good Idea At The Time.  If I re-created this project today, its architecture would be quite different.)

**Comment by zbeekman** [comment]
> As for Makefiles... well, this package builds on OSX, FreeBSD, Linux, and Solaris, so there's that...

Sure, I'm not saying you need to completely remove the functionality from the makefile, rather we need a way to guarantee that typing `make py` and `make install` will never download arbitrary resources in an unsafe way. Adding some variable that you can pass to `make` might work. (`make DISABLE_DOWNLOADS=1 install` or similar) Let's get some input from others in @Homebrew/core to see what they think.

File: Formula/astrometry-net.rb
Comment (dstndstn) [reply Formula/astrometry-net.rb]: 1. These targets are not hit during a standard "make"; "make py"; "make install" sequence.  They are there for convenience for more advanced users doing unusual things with the codebase.

2. Similarly, the python code that calls wget is not called during "vanilla" operations.

3. Indeed you are correct, curl is the default, so we can remove the wget dependency.

Comment (dstndstn) [reply Formula/astrometry-net.rb]: 1. But it turns out that some of these URL have even web-rotted away, so I am commenting-out those lines of the Makefile.  I'm also doing some work on the Makefiles to support faster parallel builds, so I can cut a new release before we merge this formula.

**Comment by dstndstn** [comment]
@zbeekman Thanks for your comments.  I removed the dependency on wget, and also commented-out all the wgets from Makefiles, in the new astrometry.net-0.76 release.

**Comment by zbeekman** [comment]
> @zbeekman Thanks for your comments. I removed the dependency on wget, and also commented-out all the wgets from Makefiles, in the new astrometry.net-0.76 release.

Great! Thanks so much. It's pretty funny I just tried to push to this PR, but you beat me to it by less than 30 seconds. Haha! Once CI passes I'll give it a final review. Unless unexpected issues come up I think this looks good and plan on :ship:ing it!

Moving forwards I *highly* recommend using secure downloads to help prevent MITM attacks and comparing downloaded assets with strong cryptographic checksums to verify the contents have not changed when building software or automatically fetching assets & data on the web.

**Comment by dstndstn** [comment]
@BrewTestBot could you please work your magic?

**Comment by dstndstn** [comment]
@BrewTestBot test this please

File: Formula/astrometry-net.rb
Comment (dstndstn) [reply Formula/astrometry-net.rb]: Well, what it does is pattern-matches a given image against a database of the sky, so a real test requires some data.  We do ship enough in the download to do a minimal test.  These files are not installed, so is it possible to access the un-tarred software directory in the 'test' block?

**Comment by dstndstn** [comment]
@BrewTestBot test this please (your last build on mojave_testing failed for internal reasons)

**Comment by zbeekman** [comment]
@dstndstn brew test bot listens to maintainers, not contributors 😉

We should be performing a better test; I'm trying to recall the correct way to pull data from the source directory for testing... So we may as well hold off until I can dig that up, or someone else beats me to it.

**Comment by dstndstn** [comment]
For what it's worth, here's the test we do in the Travis-ci build:

    - build-astrometry-index -d 3 -o index-9918.fits -P 18 -S mag -B 0.1 -s 0 -r 1 -I 9918 -M -i demo/tycho2-mag6.fits
    - echo -e 'add_path .\ninparallel\nindex index-9918.fits' > 99.cfg
    - solve-field --config 99.cfg demo/apod4.jpg  --continue

File: Formula/astrometry-net.rb
Comment (javian) [reply Formula/astrometry-net.rb]: No the files are not accessible since the test method occurs after the build has been completed and on whatever is in the bottle after its done. You can for instance run tests on your own machine when downloading a bottle without access to anything left behind after the install method has run so it would have to accommodate for that.

**Comment by javian** [comment]
That looks perfect assuming that the processing time is relatively short.

**Comment by dstndstn** [comment]
I just realized that both the required data files are actually installed in the "examples/" directory =)

**Comment by zbeekman** [comment]
Error during testing. Do the tests work after installing and running locally for you, @dstndstn? Also, there's an indentation issue flagged by `brew audit`.

```
==> /usr/local/Cellar/astrometry-net/0.76/bin/solve-field --config 99.cfg /usr/local/Cellar/astrometry-net/0.76/examples/apod4.jpg --continue --dir .
Reading input file 1 of 1: "/usr/local/Cellar/astrometry-net/0.76/examples/apod4.jpg"...
Traceback (most recent call last):
  File "/usr/local/Cellar/astrometry-net/0.76/bin/image2pnm", line 14, in <module>
    from astrometry.util.image2pnm import main
ImportError: No module named astrometry.util.image2pnm
augment-xylist.c:589:backtick Failed to run command: /usr/local/Cellar/astrometry-net/0.76/bin/image2pnm --infile /usr/local/Cellar/astrometry-net/0.76/examples/apod4.jpg --uncompressed-outfile /tmp/tmp.uncompressed.XwQ8f3 --outfile /tmp/tmp.ppm.cGAC6y --ppm --mydir /usr/local/Cellar/astrometry-net/0.76/bin/solve-field
 ioutils.c:567:run_command_get_outputs Command failed: return value 1
Error: astrometry-net: failed
Failed executing: /usr/local/Cellar/astrometry-net/0.76/bin/solve-field --config 99.cfg /usr/local/Cellar/astrometry-net/0.76/examples/apod4.jpg --continue --dir .
```

**Comment by dstndstn** [comment]
Okay, now it's failing for a slightly better reason -- missing a python fits reader (pyfits, fitsio, or astropy).  Formerly, this formula installed pyfits, but really that's the worst of the three options.  Both astropy and fitsio are pip-installable.  astropy is a pretty big package and fairly popular, while I think fitsio is technically better.  Is there a good way to ensure a pip-installable package is installed?  (I read about installing into a vendor directory in the Python for Formula Authors page, but... would rather not...)
https://docs.brew.sh/Python-for-Formula-Authors

File: Formula/astrometry-net.rb
Comment (javian) [inline Formula/astrometry-net.rb]: Great! Can you also make sure that these lines doesn't surpass the 80 character limit ? Thanks!

Comment (dstndstn) [reply Formula/astrometry-net.rb]: sure, done :)

**Comment by zbeekman** [comment]
> Okay, now it's failing for a slightly better reason -- missing a python fits reader (pyfits, fitsio, or astropy).

Is a python fits reader a general requirement of astrometry-net? If so, then maybe we should vendor fitsio? I see that cfitsio is available in homebrew, but presumably that doesn't ship with python bindings.

This would involve creating a resource block, which, for most other languages we're trying to move away from, but AFAICT is still an acceptable practice in Python, where required...

@dstndstn is the functionality significantly degraded without a python fits reader? Or should we ship this as is?

**Comment by dstndstn** [comment]
Right, the purpose of the fitsio package is to provide python bindings for cfitsio.

Yes, by default the main solve-field program calls two python programs to clean up the inputs, so without a fits reader (and without extra command-line args to turn off these cleanup routines), it would crash out of the box (hence the --no-remove-lines and --uniformize 0 args in the tests).  Thinking on it, I could implement these in C instead without too much work; or recover with more grace from failures of these programs.

**Comment by zbeekman** [comment]
> Yes, by default the main solve-field program calls two python programs to clean up the inputs, so without a fits reader (and without extra command-line args to turn off these cleanup routines), it would crash out of the box (hence the --no-remove-lines and --uniformize 0 args in the tests). Thinking on it, I could implement these in C instead without too much work; or recover with more grace from failures of these programs.

From the sound of it, I think that we should either:

1. Vendor a python fitsio implementation (it sounds like your preference fitsio)
2. Implement the functionality upstream in astrometry-net natively (in C etc.)
3. Make fitsio "more" optional (i.e. recover gracefully and process w/o cleanup)

Since we're removing options from formulae, I think that 1 or 2 is the best solution, 2 is probably most robust, and we'll avoid the (becoming undesirable) vendoring of dependencies through resource blocks.

**Comment by dstndstn** [comment]
I would like to do (1) as a near-term solution, and I will plan to implement (2) upstream longer-term.

I can read the Python for Formula Authors page on implementing (1), but would happily accept any advice or help!

**Comment by zbeekman** [comment]
Dustin: FYI I haven’t forgotten about this. Can you please point me to the
PyPI page of the FITS package you want to use? I will look into creating a
resource block. Alternatively take a look at a formula like FORD (`brew cat
ford` for an example of how python modules are vendored with `resource`
blocks.

**Comment by dstndstn** [comment]
Thanks for your note -- I've been busy (just started a new job) so haven't had a chance to work on this.
Here's fitsio:
https://pypi.org/project/fitsio/

**Comment by dstndstn** [comment]
@zbeekman Hello again, nearly 2 months later!  I just tried vendoring the 'fitsio' package, and adjusting the tests so that it would be used.  I would have expected that I also have to do something in my code to actually *use* the vendored virtualenv... hoping that magic occurs and the brew test bot finishes successfully!

**Comment by zbeekman** [comment]
@dstndstn I was just thinking about this PR and meant to circle back to it, good timing!

It looks like the vendoring of the deps worked. I think we're also moving towards the more pythonic way of allowing a standard, pythonic build system (e.g., `setup.py`) handle dependency installations as long as they're sane. (e.g., `requirements.txt with `package >= version` etc.)

Any way, I'm happy to keep this as is, with vendored resources.

Right now there are two problems:

1. http://astrometry.net is unreachable, so audit fails.
2. There is a test failure. I'm not sure what to make of it:
```
Traceback (most recent call last):
  File "/usr/local/Cellar/astrometry-net/0.76/bin/removelines", line 15, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.7/site-packages/astrometry/util/removelines.py", line 78, in main
    cut=opt.cut, ext=opt.ext)
  File "/usr/local/lib/python3.7/site-packages/astrometry/util/removelines.py", line 40, in removelines
    T = fits_table(infile, lower=False)
  File "/usr/local/lib/python3.7/site-packages/astrometry/util/fits.py", line 672, in fits_table
    pf = pyfits.open(dataorfn, memmap=mmap)
AttributeError: 'NoPyfits' object has no attribute 'open'
augment-xylist.c:589:backtick Failed to run command: /usr/local/Cellar/astrometry-net/0.76/bin/removelines /tmp/tmp.xyls.Eu3Adr /tmp/tmp.removelines.37H9ik
 ioutils.c:567:run_command_get_outputs Command failed: return value 1
Error: astrometry-net: failed
```

**Comment by dstndstn** [comment]
@zbeekman Thanks for your comments.  The astrometry.net site is currently down due to a hardware failure, so I temporarily changed the homepage in the formula.  I also revamped the installation process as it connects to the vendored packages.  audit and brew test bot builds look clean!

File: Formula/astrometry-net.rb
Comment (zbeekman) [inline Formula/astrometry-net.rb]: These comments look outdated. If so, can you please remove them? Any deficiencies with the packaging/installation of software should be reported as an issue to upstream (i.e., your astrometry repository). If it's actually pertinent, comments at work arounds should link to the upstream issue and indicate what needs to change when upstream mints a new release resolving the issue.

To me, it looks like you can just remove both of these comments.

File: Formula/astrometry-net.rb
Comment (zbeekman) [inline Formula/astrometry-net.rb]: Can you please try this instead of using `ln_sf`:

```suggestion
    bin.install_symlink libexec/"lib/python#{xy}/site-packages/astrometry/blind/plotann.py"
```

Also, the comment here should be changed to link to the upstream issue, indicating when it can be removed.

e.g.,
```
Work around for https://github.com/dstndstn/issues/###
```

**Review by zbeekman — CHANGES_REQUESTED** [review]

**Comment by dstndstn** [comment]
Hi @zbeekman , thanks for all your attention!  This work-around is required because the Makefile wants to create a link from bin/plotann.py to ../lib/python/astrometry rather than the ../libexec/lib/python3.7/...... directory that we are putting vendored python stuff into.  I created (and fixed) the upstream issue, but don't really want to cut a new release just for this one-char change...

Your suggested fix for creating the symlink works nicely -- thanks!

File: Formula/astrometry-net.rb
Comment (dstndstn) [reply Formula/astrometry-net.rb]: oops, should have replied here rather than below!  Once this FIXME is done, there will not be a need to special-case the symlink for plotann.py; the Makefile will just do the right thing with that PY_BASE_LINK_DIR variable set.

File: Formula/astrometry-net.rb
Comment (zbeekman) [reply Formula/astrometry-net.rb]: @dstndstn can you please completely remove these comments? They are too verbose. Let's add the issue number and some concise details near the symlink creation.

File: Formula/astrometry-net.rb
Comment (zbeekman) [inline Formula/astrometry-net.rb]: Please replace:
```
    # FIXME: work around for https://github.com/dstndstn/astrometry.net/issues/142
    # Once the next release comes out, remove the following two lines.
```
with
```
    # Work around for https://github.com/dstndstn/astrometry.net/issues/142
    # On the next release, remove the following two lines & add `ENV["PY_BASE_LINK_DIR"] = ...`
```

**Review by zbeekman — CHANGES_REQUESTED** [review]

File: Formula/astrometry-net.rb
Comment (dstndstn) [reply Formula/astrometry-net.rb]: Sure thing!

**Review by zbeekman — APPROVED** [review]
LGTM

Gonna leave open for a bit incase any other @Homebrew/core folks want to give this a once over.

File: Formula/astrometry-net.rb
Comment (sjackman) [inline Formula/astrometry-net.rb]: You have both `depends_on "numpy"` and this `numpy` resource. Can this resource be removed?

File: Formula/astrometry-net.rb
Comment (sjackman) [inline Formula/astrometry-net.rb]: It'd be great to test that the expected output files exist after running this command.
```ruby
assert_predicate testpath/"output-filename-goes-here", :exist?
```

**Review by sjackman — CHANGES_REQUESTED** [review]

File: Formula/astrometry-net.rb
Comment (zbeekman) [reply Formula/astrometry-net.rb]: Good catch.

Comment (dstndstn) [reply Formula/astrometry-net.rb]: We're using virtualenv to vendor a package (fitsio) that depends on numpy.  If we have 'numpy' as a dependency, does that package become available within the virtualenv?  Or does it have to get added to the virtualenv?  (Or do we have to use "virtualenv --system-site-packages"?)

Comment (sjackman) [reply Formula/astrometry-net.rb]: Yes, it looks like `virtualenv --system-site-packages` would you what you want.

Rather than using `virtualenv` to install a resource, another option is to use `pip3` to install `fits`.
```ruby
xy = Language::Python.major_minor_version "python3"
ENV.prepend_path "PYTHONPATH", libexec/"lib/python#{xy}/site-packages"
system "pip3", "install", "--prefix=#{libexec}", "fitsio=0.9.11"
…
bin.env_script_all_files libexec/"bin", :PYTHONPATH => ENV["PYTHONPATH"]
```

Comment (dstndstn) [reply Formula/astrometry-net.rb]: I would love that pip3 solution; way easier, in my opinion.

Does the env_script_all_files magic edit scripts to modify the pythonpath at the beginning of the scripts in the bin directory?  (assuming shell scripts?  Some of the scripts in bin/ (not all ending in .py) are python scripts starting with "#! /usr/bin/env python3" or the equivalent.

But isn't this pip solution somewhat contrary to the "Python for Formula Authors" advice to install 'applications' in a virtualenv?

Comment (sjackman) [reply Formula/astrometry-net.rb]: > But isn't this pip solution somewhat contrary to the "Python for Formula Authors" advice to install 'applications' in a virtualenv?

I'll defer to a Homebrew/core maintainer that has more experience than myself. I've used the above `pip3` snippet a fair bit in Brewsci/bio.

Comment (zbeekman) [reply Formula/astrometry-net.rb]: @commitay, @javian, @fxcoudert or anyone else from @Homebrew/core with more python package experience than me, could you please weigh in here when you have a chance? Thanks.

Comment (sjackman) [reply Formula/astrometry-net.rb]: > Does the env_script_all_files magic edit scripts to modify the pythonpath at the beginning of the scripts in the bin directory?

It moves all executables found in `bin` to `libexec/bin`, then creates a shell script in `bin` that sets `PYTHONPATH`  and calls the real executable that has been moved to `libexec/bin`.

Comment (dstndstn) [reply Formula/astrometry-net.rb]: Oh, cool, that's handy.

Thanks for your comments and advice here!

File: Formula/astrometry-net.rb
Comment (dstndstn) [reply Formula/astrometry-net.rb]: Thanks, I added a couple of tests like this to assert the existence of the (empty) file we write to indicate success, and the primary result file (.wcs).

Comment (sjackman) [reply Formula/astrometry-net.rb]: Great!

**Comment by zbeekman** [comment]
@dstndstn I hope you don't mind: I want to put this PR to bed, so I:

- cleaned up the remaining todo items
- rebased onto homebrew-core:master
- Squashed all the commits
- Force pushed back to your branch (I hope this doesn't cause you issues, if it does let me know and I can force push to restore the old branch to the way it was. Or you can do a `git fetch` followed by a `git reset --hard <name of your fork's origin>/migrate-astrometry-net`

Pending successful CI & bottling I plan to merge this, unless other @homebrew/core maintainers have any objections or further input.

**Comment by dstndstn** [comment]
I don't mind at all -- this is awesome!  Thanks again for all your work on this.

**Comment by zbeekman** [comment]
If you can test locally to ensure proper functionality that would be great.

**Review by sjackman — APPROVED** [review]

**Comment by dstndstn** [comment]
brew install astrometry-net worked for me!

**Comment by sjackman** [comment]
Was anything special required to make the `fitsio` in `virtualenv` use the brewed `numpy`? I don't see any special hacks in the `astrometry-net` formula.

**Comment by dstndstn** [comment]
hi @sjackman I believe it is installing a new numpy in the virtualenv

```
==> Downloading https://files.pythonhosted.org/packages/4e/8b/75469c270ac544265f0020aa7c4ea925c5284b23e445cf3aa8b99f662690/virtualenv-16.1.0.tar.gz
...
==> python3 -c import setuptools... --no-user-cfg install --prefix=/private/tmp/astrometry-net--homebrew-virtualenv-20181119-26641-1alhllo/target --install-scripts=/private/tmp/a
strometry-net--homebrew-virtualenv-20181119-26641-1alhllo/target/bin --single-version-externally-managed --record=installed.txt
...
==> python3 -s /private/tmp/astrometry-net--homebrew-virtualenv-20181119-26641-1alhllo/target/bin/virtualenv -p python3 /usr/local/Cellar/astrometry-net/0.76/libexec
Using base prefix '/usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7'
New python executable in /usr/local/Cellar/astrometry-net/0.76/libexec/bin/python3.7
Also creating executable in /usr/local/Cellar/astrometry-net/0.76/libexec/bin/python
Installing setuptools, pip, wheel...
done.
Running virtualenv with interpreter /usr/local/opt/python/bin/python3
==> Downloading https://files.pythonhosted.org/packages/83/6b/d03277eacf113697675cd659086a4dcf9472108e2f1a83884c0271bdca46/numpy-1.15.3.zip
...
==> Downloading https://files.pythonhosted.org/packages/9c/cb/f52534b71f4d99916723af2994898904015b9a1bf0286a165182d0374bbf/fitsio-0.9.11.tar.gz
...
```

**Comment by sjackman** [comment]
Ah, yes, it would. I'd prefer not to install a duplicate `numpy` in the virtualenv and instead use the brewed `numpy`. I don't feel strongly enough to block this PR on my preference. How do you @dstndstn and @zbeekman feel about it?

**Comment by dstndstn** [comment]
I feel like the whole vendored environment is overkill and I would rather just have users do a "pip install fitsio" :)

**Comment by sjackman** [comment]
I felt the same way originally, because that's how it used to be, but since moving to vendouring the Python dependencies, I see now that it's way more reliable. I personally prefer to use `pip3 install --prefix=#{libexec} foo bar baz` in the formula to install the dependencies rather than resources and virtualenv. You can specify particular versions to install.

**Comment by sjackman** [comment]
@zbeekman The virtualenv by default does not have access to the brewed `numpy` package. `fitsio` has a dependency on `numpy`, so installing the `fitsio` installs its unsatisfied dependency `numpy`. We could either configure virtualenv to have access to the system packages using `virtualenv --system-site-packages`, or we could use `pip3 install --prefix=#{libexec} fitsio` rather than using virtualenv, or accept this duplication of `numpy`.

**Comment by dstndstn** [comment]
@zbeekman Sorry!, I was wrong -- indeed I was not using an up-to-date version of the formula (I was trying to work on too many things today and forgot that my ~/homebrew-core was a symlink into /usr/local/Homebrew, so moving ~/homebrew-core out of the way and doing a fresh 'git clone' does nothing!)

You are correct, it does not install numpy yet still has access to the brewed one.

Install:
```
==> Installing astrometry-net
==> Downloading https://github.com/dstndstn/astrometry.net/releases/download/0.76/astrometry.net-0.76.tar.gz
Already downloaded: /Users/dstn/Library/Caches/Homebrew/downloads/ca323815ac71a727bb89fea7d9d1938a6d9feeef23f3283d61d1b08351cf1307--astrometry.net-0.76.tar.gz
==> Downloading https://files.pythonhosted.org/packages/4e/8b/75469c270ac544265f0020aa7c4ea925c5284b23e445cf3aa8b99f662690/virtualenv-16.1.0.tar.gz
Already downloaded: /Users/dstn/Library/Caches/Homebrew/downloads/bda8bcde19a275fd9c84a7ec957593976e90dfd0e461a87f5698f1d2286c0197--virtualenv-16.1.0.tar.gz
==> python3 -c import setuptools... --no-user-cfg install --prefix=/private/tmp/astrometry-net--homebrew-virtualenv-20181119-55045-1b1vy6l/target --install-scripts=/private/tmp/a
==> python3 -s /private/tmp/astrometry-net--homebrew-virtualenv-20181119-55045-1b1vy6l/target/bin/virtualenv -p python3 /usr/local/Cellar/astrometry-net/0.76/libexec
==> Downloading https://files.pythonhosted.org/packages/9c/cb/f52534b71f4d99916723af2994898904015b9a1bf0286a165182d0374bbf/fitsio-0.9.11.tar.gz
Already downloaded: /Users/dstn/Library/Caches/Homebrew/downloads/739f19b7e0abb5bd0e3a1750abd75c2d415091beb71037f27a50160baf668e5a--fitsio-0.9.11.tar.gz
==> /usr/local/Cellar/astrometry-net/0.76/libexec/bin/pip install -v --no-deps --no-binary :all: --ignore-installed /private/tmp/astrometry-net--fitsio-20181119-55045-1se5467/fit
==> make
==> make py
==> make install
🍺
```

After:
```
> PYTHONPATH= /usr/local/Cellar/astrometry-net/0.76/libexec/bin/python -c "import fitsio; import numpy; print(fitsio.__file__); print(numpy.__file__)"
/usr/local/Cellar/astrometry-net/0.76/libexec/lib/python3.7/site-packages/fitsio/__init__.py
/usr/local/opt/numpy/lib/python3.7/site-packages/numpy/__init__.py
```

So this looks great to me!

**Comment by sjackman** [comment]
Ah, good! I'm glad to hear that I'm wrong, and that `fitsio` will in fact used the brewed `numpy` by default.

**Comment by zbeekman** [comment]
> You are correct, it does not install numpy yet still has access to the brewed one.

😅

**Comment by dstndstn** [comment]
The build failures do not seem to be related to this formula, they're about failure to clean up fully?

**Comment by Moisan** [comment]
@BrewTestBot test this please

**Comment by dstndstn** [comment]
Yay, buildbot finished successfully!

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
Thanks @dstndstn for the pull request and your patience!

**Comment by zbeekman** [comment]
Yes, thanks for being patient, and thank you for your contribution to Homebrew!

**Pull request closed without merging**
